protostar/commands/test/test_command.py

```diff
@@ -134,6 +150,14 @@ async def run(self, args) -> TestingSummary:
             seed=args.seed,
             slowest_tests_to_report_count=args.report_slowest_tests,
         )
+        failed_tests_paths = [
+            str(failed_test.file_path.absolute()) for failed_test in summary.failed
+        ]
+        protostar_cache.persist(
+            self._project_root_path,
+            "test_results",
+            {"failed_tests": failed_tests_paths},
+        )
```

Comment:
Currently, running integration tests will modify `.protostar_cache`.

`protostar_cache` should be easily replaceable for testing purposes. I usually do it by using classes and composition. 

- [ ] avoid changing .`protostar_cache` when running integration tests

Reply:
> Currently, running integration tests will modify `.protostar_cache`.

Are you sure about this? I might've missed something but I just ran integration tests and the file's not been created. Looking at the code, I don't see a place other than `cli_app.py` where `TestCommand::run` would be invoked, and we're dealing with the cache only in the `run` method.

> `protostar_cache` should be easily replaceable for testing purposes. I usually do it by using classes and composition.

I can wrap it into a class, that's not a problem.

Reply:
> Are you sure about this? I might've missed something but I just ran integration tests and the file's not been created. Looking at the code, I don't see a place other than cli_app.py where TestCommand::run would be invoked, and we're dealing with the cache only in the run method.

Yep, you right. I thought this code was in `TestCommand::test` method.

Usually we use`Command::run` to passing `args` as typed parameters to the proper method (`test` in this case). This code operates on untyped args and data (`targets`, `last_failed`, `previous_results`).